tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MatchTypeSuite.scala

```diff
@@ -274,9 +274,11 @@ class MatchTypeSuite extends BaseDottySuite {
                 Type.Name("L"),
                 List(
                   TypeCase(
-                    Type.Apply(
-                      Type.Name("Foo"),
-                      List(Type.AnonymousParam(None))
+                    Type.AnonymousLambda(
```

Comment:
Isn't this actually just a placholder? `Foo[_] ` it's not a lambda

Reply:
but for scala32, `_` is not a wildcard anymore, hence `F[_]` is actually only valid as a lambda.

Reply:
I don't think it happened yet despite what is in the plans.

Reply:
IF I reprint the statement with Dotty I get back:
```
object match_types:
  type Combine[L, R] = L match
    case Foo[?] => L
    case Bar[?] => L
    case _ => R
    case ? => L
```

Reply:
> I don't think it happened yet despite what is in the plans.

in that case, `AnonymousParam` that we added in the previous commit is also invalid (that and AnonymousLambda should go together).

we might need to disable this field in scala3.2 and just set it in tests specifically, or add a new scala3.3 dialect:
```
    // Dotty uses `_` for placeholder for types since 3.2
    val allowUnderscoreAsTypePlaceholder: Boolean,
```

Reply:
> IF I reprint the statement with Dotty I get back:
> 
> ```
> object match_types:
>   type Combine[L, R] = L match
>     case Foo[?] => L
>     case Bar[?] => L
>     case _ => R
>     case ? => L
> ```

so `_` is still a wildcard, not a placeholder, then. is it with unreleased nightly 3.2, or a prior version?

Reply:
hmm, looks like 3.2.0 has already been released, and doesn't mention this. should we attach `allowUnderscoreAsTypePlaceholder` to a new 3.3 dialect instead?

Reply:
> in that case, AnonymousParam that we added in the previous commit is also invalid (that and AnonymousLambda should go together).

Ach sorry, I didn't realize that :fearful: 

> hmm, looks like 3.2.0 has already been released, and doesn't mention this. should we attach allowUnderscoreAsTypePlaceholder to a new 3.3 dialect instead?

I think that might be good enough for now.

Reply:
I will follow up with the compiler team to see when they plan to deprecate it.

Reply:
I didn't get any response, but maybe it's fine to disable it for now until we know in which dilect it will be introduced?

Reply:
will do, thank you!

Reply:
last question: do you know if this `_` as lambda placeholder (not wildcard) is supported in any of the scala2[source3] variants?

Reply:
There is apparently the flag `-source:future` that can be used, but that's apparently a bunch different features, which will be introduced later on. We could possibly introduce the dialect `Scala3Future` that could contain them though I don't know if there is a list of all the future features.